compiler/stable_mir/src/compiler_interface.rs

```diff
@@ -62,15 +61,6 @@ pub trait Context {
     /// Returns the name of given `DefId`
     fn def_name(&self, def_id: DefId, trimmed: bool) -> Symbol;
 
-    /// Return attributes with the given attribute name.
-    ///
-    /// Single segmented name like `#[inline]` is specified as `&["inline".to_string()]`.
-    /// Multi-segmented name like `#[rustfmt::skip]` is specified as `&["rustfmt".to_string(), "skip".to_string()]`.
-    fn get_attrs_by_path(&self, def_id: DefId, attr: &[Symbol]) -> Vec<Attribute>;
-
-    /// Get all attributes of a definition.
-    fn get_all_attrs(&self, def_id: DefId) -> Vec<Attribute>;
-
```

Comment:
Please don't remove this. Tools depend on this to retrieve information about tool attributes. Let's talk offline to see what can be done.

Reply:
Tools will have to change the API they use; I think this is one of these changes where it's going to be an inappropriate amount of work to keep the old API around.

Reply:
Sure. But right now there's no way to get that information.

Reply:
fixed in the latest changes, for specifically tool attributes these functions are available again under a different name (as discussed)